execution-plane/arcon/arcon_state/src/lib.rs

```diff
@@ -0,0 +1,364 @@
+// Copyright (c) 2020, KTH Royal Institute of Technology.
+// SPDX-License-Identifier: AGPL-3.0-only
+#![feature(associated_type_defaults)]
```

Comment:
Are these features gonna hit stable anytime soon?

Reply:
https://github.com/rust-lang/rust/issues/29661 no idea. We don't necessarily need this, it just makes the code slightly shorter. We're not targeting stable anyway, so I see no good reason to remove this. But if we wanted to, it's easy to remove.

Reply:
While we don't exactly need to run on stable right now, I think it would be nice to aim towards this in the future, so that is why I am asking. Nightly features are fine, but would like to avoid to rely  on features that may take 2-3 years to get merged.

Reply:
Oh, the comment I left here is just for `associated_type_defaults`. The `const_generics` feature is to be able to use fixed-length arrays as item keys or namespaces in state backends. I don't know if we use this anywhere, but it seems useful. If we want to get rid of this, I can put it behind a feature flag and check if the code still compiles.

Reply:
It is fine :) Just something to have in mind.